src/SDKs/Azure.Core/data-plane/Azure.Core.Tests/SocketTransport/Sequence.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Buffers;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Buffers
+{
+    /// <summary>
+    /// Sequence<typeparamref name="T"/> is a builder of a sequence of buffers/arrays rented from a buffer pool
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <remarks>Be very careful when disposing Sequence<typeparamref name="T"/>. Calling a dispose on a sequence copy will result in pool corruption.</remarks>
+    struct Sequence<T> : IBufferWriter<T>
```

Comment:
Possibly this can be moved back into labs after the sockets transport refactoring.

Reply:
Let's track both with the same issue (https://github.com/Azure/azure-sdk-for-net/issues/5219)